types/component/dispatcher/dispatcher.go

```diff
@@ -30,57 +30,60 @@ func (a *AmountDispatched) IsPositive() bool {
 	return a.Incoming.IsPositive() || a.Outgoing.IsPositive()
 }
 
-func NewAmountDispatched(
-	incoming math.Int,
-	outgoing math.Int,
-) *AmountDispatched {
-	return &AmountDispatched{
-		Incoming: incoming,
-		Outgoing: outgoing,
+func (a DispatchedAmountEntry) Validate() error {
+	if a.Denom == "" {
+		return errorsmod.Wrap(core.ErrValidation, "cannot set empty denom")
 	}
-}
 
-type ChainAmountDispatched struct {
-	ccID             core.CrossChainID
-	amountDispatched AmountDispatched
-}
+	if a.SourceId == nil {
+		return errorsmod.Wrap(core.ErrValidation, "missing source cross-chain ID")
+	}
+	if err := a.SourceId.Validate(); err != nil {
+		return errorsmod.Wrap(err, "failed to create source cross-chain ID")
+	}
 
-func NewChainAmountDispatched(
-	ccID core.CrossChainID,
-	amountDispatched AmountDispatched,
-) *ChainAmountDispatched {
-	return &ChainAmountDispatched{
-		ccID:             ccID,
-		amountDispatched: amountDispatched,
+	if a.DestinationId == nil {
+		return errorsmod.Wrap(core.ErrValidation, "missing destination cross-chain ID")
 	}
-}
 
-func (cad *ChainAmountDispatched) CrossChainID() core.CrossChainID {
-	return cad.ccID
-}
+	if err := a.DestinationId.Validate(); err != nil {
+		return errorsmod.Wrap(err, "failed to create destination cross-chain ID")
+	}
 
-func (cad *ChainAmountDispatched) AmountDispatched() AmountDispatched {
-	return cad.amountDispatched
-}
+	if a.AmountDispatched.Incoming.IsNegative() || a.AmountDispatched.Outgoing.IsNegative() {
+		return errorsmod.Wrap(core.ErrValidation, "cannot set negative amounts")
+	}
+
+	if !a.AmountDispatched.Incoming.IsPositive() && !a.AmountDispatched.Outgoing.IsPositive() {
+		return errorsmod.Wrap(
+			core.ErrValidation,
+			"cannot set incoming and outgoing amounts equal to zero",
+		)
+	}
 
-type TotalDispatched struct {
-	chainsAmount map[string]ChainAmountDispatched
+	return nil
 }
 
-func NewTotalDispatched() *TotalDispatched {
-	return &TotalDispatched{
-		chainsAmount: make(map[string]ChainAmountDispatched),
+func (c DispatchCountEntry) Validate() error {
+	if c.Count == 0 {
+		return errorsmod.Wrap(core.ErrValidation, "cannot set zero count")
```

Comment:
nit: I believe we decided to wrap the return of `Validate` method with the `ErrValidation` and to not use it inside the methods. 

```suggestion
		return errorsmod.Wrap(core.ErrNilPointer, "cannot set zero count")
```